improver/psychrometric_calculations/hail_size.py

```diff
@@ -0,0 +1,574 @@
+
+            pressure_diff = abs(grid - press.data)
+            indices = np.nanargmin(pressure_diff, axis=0)
+            lat, long = indices.shape
+            lat, long = np.ogrid[:lat, :long]
+            if variable is None:
+                variable = var.data[indices, lat, long]
+            else:
+                variable.append(var.data[indices, lat, long])
+
+        variable_cube = pressure.copy(data=variable)
+        enforce_coordinate_ordering(variable_cube, coord_order)
+        enforce_coordinate_ordering(pressure, coord_order)
+        return variable_cube.data
+
+    def pressure_grid(self, variable_on_pressure: Cube) -> np.ndarray:
+        """Creates a pressure grid of the same shape as variable_on_pressure cube.
+        It is populated at every grid square and for every realization with
+        a column of all pressure levels taken from variable_on_pressure's pressure coordinate
+
+        Args:
+            Variable_on_pressure
+                Cube of some variable with pressure levels
+        Returns:
+            An n dimensional array containing at every grid square and for every realization
+            a column of all pressure levels taken from variable_on_pressure's pressure coordinate
+        """
+
+        required_shape = variable_on_pressure.shape
+        pressure_points = variable_on_pressure.coord("pressure").points
+        (pressure_axis,) = variable_on_pressure.coord_dims("pressure")
+        pressure_shape = np.ones_like(required_shape)
+        pressure_shape[pressure_axis] = required_shape[pressure_axis]
+        pressure_array = np.broadcast_to(
+            pressure_points.reshape(pressure_shape), required_shape
+        )
+        return pressure_array
+
+    def extract_pressure_at_268(self, temperature_on_pressure: Cube) -> (CubeList):
+        """Extracts the pressure level where the environment
+        temperature first drops below -5 degrees (268.15K) starting at a pressure value
+        near the surface and ascending in altitude from there. It also produces
+        the environment temperature at that pressure value
+
+        Args:
+            temperature_on_pressure
+                A cube of temperature on pressure levels
+        Returns:
+            A list of two cubes containing a cube of the environment pressure at 268.15K
+            and a cube of the temperature at that pressure value
+        """
+
+        pressure_template = next(temperature_on_pressure.slices_over(["pressure"]))
+        pressure_template.rename("pressure_of_atmosphere_at_268.15K")
+        pressure_template.units = temperature_on_pressure.coord("pressure").units
+        pressure_template.remove_coord("pressure")
+
+        temperature_template = next(temperature_on_pressure.slices_over(["pressure"]))
+        temperature_template.rename("temperature_of_atmosphere_at_268.15K")
+        temperature_template.remove_coord("pressure")
```

Comment:
These do need these to be cubes because they go into the HumidityMixingRatio plugin and that needs cubes as inputs. If Im going to be using cubes anyway I felt they might as well have correct metadata

Reply:
I didn't spot that. Fair enough.